src/CommunityToolkit.HighPerformance/Streams/ReadOnlySequenceStream.cs

```diff
@@ -0,0 +1,303 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace CommunityToolkit.HighPerformance.Streams;
+
+/// <summary>
+/// A <see cref="Stream"/> implementation wrapping a <see cref="ReadOnlySequence{T}"/> of <see cref="byte"/> instance.
+/// </summary>
+internal sealed partial class ReadOnlySequenceStream : Stream
+{
+    /// <summary>
+    /// The <see cref="ReadOnlySequence{T}"/> instance currently in use.
+    /// </summary>
+    private readonly ReadOnlySequence<byte> source;
+
+    /// <summary>
+    /// The current position within <see cref="source"/>.
+    /// </summary>
+    private int position;
```

Comment:
Should this be a `long`?

Reply:
Yes!

I had just copied what's on `MemoryStream<TSource>`.

Changed!

Reply:
I think you accidentally pushed one of your old commits again, I see a bunch of stuff in here that I had removed. Can you drop the second to last commit? Otherwise I can fix this tomorrow and just force push again.